### PR TITLE
Replace autogen.py with something simpler.

### PR DESCRIPTION
**Pull request by le-jzr**
Instead of generating headers for certain structures, the headers are written manually and we automate checking correctness instead. Checking is performed by generating a C source with a bunch of static asserts, using a simple awk script. This is then treated as a normal source file.

The primary motivation for this change is to reduce the complexity of the build process. Also, the `.ag` files we used previously are more difficult to understand than regular C code, and at least one IDE (GNOME Builder) completely refuses to open them.

### CODE REVIEW

File: kernel/arch/amd64/include/arch/boot/memmap_struct.h [hunk 1]
```diff
@@ -0,0 +1,46 @@
...
+ * IN NO EVENT SHALL THE AUTHOR BE LIABLE FOR ANY DIRECT, INDIRECT,
+ * INCIDENTAL, SPECIAL, EXEMPLARY, OR CONSEQUENTIAL DAMAGES (INCLUDING, BUT
+ * NOT LIMITED TO, PROCUREMENT OF SUBSTITUTE GOODS OR SERVICES; LOSS OF USE,
+ * DATA, OR PROFITS; OR BUSINESS INTERRUPTION) HOWEVER CAUSED AND ON ANY
+ * THEORY OF LIABILITY, WHETHER IN CONTRACT, STRICT LIABILITY, OR TORT
+ * (INCLUDING NEGLIGENCE OR OTHERWISE) ARISING IN ANY WAY OUT OF THE USE OF
+ * THIS SOFTWARE, EVEN IF ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
+ */
+
+#pragma once
```
Comment (jermar) [inline kernel/arch/amd64/include/arch/boot/memmap_struct.h]: Really? :-)

Comment (le-jzr) [reply kernel/arch/amd64/include/arch/boot/memmap_struct.h]: Darn, my evil schemes foiled. :) Shall I uglify those? :-P

Comment (le-jzr) [reply kernel/arch/amd64/include/arch/boot/memmap_struct.h]: Done.

File: tools/autocheck.awk [hunk 2]
```diff
@@ -0,0 +1,38 @@
+#!/usr/bin/awk -f
+
```
Comment (jermar) [inline tools/autocheck.awk]: Missing copyright header.

Comment (le-jzr) [reply tools/autocheck.awk]: Oops, forgot.

**Pull request merged**